plugins/modules/artisan.py

```diff
@@ -0,0 +1,324 @@
+
+version_added: 8.4.0
+
+description:
+    - >
+      Artisan is the command line interface included with Laravel, the
+      PHP web application framework for artisans. The module was heavily
+      inspired by M(community.general.composer), which is typically used together with artisan.
+
+extends_documentation_fragment:
+    - community.general.attributes
+
+attributes:
+    check_mode:
+        support: full
+    diff_mode:
+        support: none
+
+options:
+    working_dir:
+        description:
+            - Directory of your project that contains the Artisan executable.
+        required: true
+        type: path
+    command:
+        description:
+            - Artisan command like V(migrate), V(clear-compiled), or V(app:custom-command).
+        required: true
+        type: str
+    options:
+        description:
+            - Command options like seed or force without leading hyphens.
+        required: false
+        elements: str
+        default: []
+        type: list
+    args:
+        description:
+            - Command arguments, mainly in custom commands.
+        required: false
+        elements: str
+        default: []
+        type: list
+    php_path:
+        description:
+            - Path to the PHP executable on the remote host if PHP is not included in E(PATH).
+        required: false
+        type: path
+
+requirements:
+    - Laravel
+    - PHP
+
+notes:
+    - Always appended in each execution are C(--no-ansi), C(--no-interaction), and C(--force) if available.
+    - Ansible Artisan is intended primarily, but not exclusively, for production deployments.
+
+author:
+    - Ralf Langebrake (@codebarista)
+'''
+
+EXAMPLES = r'''
+# php artisan down --secret="t0pS3cRet" --refresh=60 --retry=60 --no-ansi --no-interaction
+- name: Put the application into maintenance / demo mode
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    command: down
+    options:
+      - secret="ToPs3cReT"
+      - refresh=60
+      - retry=60
+
+# php artisan storage:link --no-ansi --no-interaction --force
+- name: Connect the public/storage link to storage/app/public
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    command: storage:link
+
+# php artisan app:custom-command --no-ansi --no-interaction maybe/a/path
+- name: Run custom command with arguments
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    php_path: /usr/local/bin/php
+    command: app:custom-command
+    arguments: maybe/a/path
+  changed_when: "'modified' in app_custom_command.stdout"
+  register: app_custom_command
+
+# php artisan migrate --database=sqlite --isolated=true --seed --force
+- name: Run migrations and seed database
+  community.general.artisan:
+    working_dir: "{{ project_src }}"
+    command: "{{ item }}"
+    options:
+      - database=sqlite
+      - isolated=true
+  loop:
+    - migrate
+    - seed
+
+# php artisan up --no-ansi --no-interaction
```

Comment:
Technically, the code will generate:
```suggestion
# php {{ project_src }}/artisan up --no-ansi --no-interaction
```
If, as I suspect, `php` does not require an absolute path for its command, then I would suggest changing the `run_command` lines (253-255 at the time of writing) to:
```python
    command = '%s %s %s %s %s' % (php, "artisan", command, options, arguments)

    return module.run_command(command.strip(), cwd=module.params['working_dir'])
```
so that the current working directory of the `php` execution will be the `working_dir` ;-) and then you can get rid of the `get_artisan()` function entirely.

(Please note that I have more comments about that range of lines)